### ISSUE TITLE
create checker profiles

### ISSUE THREAD

**Issue by dkrupp** [Ericsson/codechecker#907]
Add new checker pre-selection switches to `analyze` sub-command to enable easier
checker selection for users.

Extend existing `analyze` subcommand `-e,-d` switches

to enable or disable wanted profile.

Add the following switches to `checkers` subcommand

`--list-profiles` List the available checker profiles
`--profile` <profile_name> List chekcers from the defined profile

The following profiles should be added:
`default`: The current default checker selection. All non-style, non-readability, non-product specific non-alpha checkers, with low false positive rate.
`sensitive`: Checkers in normal level + alpha checkers that do not crash and have a low false positiv rate on large open source projects
`extreme`: Enable sensitive checkers + alpha chekcers that do not crash but may have a higher (but not intolerable false positive rate).

The config file format should look like the following
```
"clangsa_checkers" : [
    {"core" : ['default', 'sensitive', 'extreme'] },
    {"security.insecureAPI.UncheckedReturn" : ['default', 'sensitive', 'extreme'] },
]
```

**Comment by Xazax-hun** [comment]
Alternatively, we could have a JSON file to describe such profiles such as:
default, sensitive, extreme, style, security etc.

Each profile could have a separate set of checks that are enabled.
We could have the following commands:
`--list-profiles` list the profiles with the checker set
`--enable-profile profile_name` enable the set of checkers in a profile

This way the users could create custom profiles if they wanted to.

What do you think?

**Comment by whisperity** [comment]
I don't think a separate opt-string would be needed for this. The `-e`/`-d` command could be extended to take not only checker names and checker groups, but severity groups and tags.

And the already existing severity file could be used to also map checkers to these tags.

**Comment by whisperity** [comment]
The problem with the optstrings in the opening post is that the "low false positive rate" is really subjective from CodeChecker's point of view, which means this will add extra lists that need be maintained. Aren't we against such?

**Comment by gyorb** [comment]
The feature was merged.
